addons/binding/org.openhab.binding.vera2/src/main/java/org/openhab/binding/vera2/handler/VeraSceneHandler.java

```diff
@@ -0,0 +1,380 @@
+            return (VeraBridgeHandler) handler;
+        } else {
+            return null;
+        }
+    }
+
+    private VeraSceneConfiguration loadAndCheckConfiguration() {
+        VeraSceneConfiguration config = getConfigAs(VeraSceneConfiguration.class);
+        if (config.getSceneId() == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Couldn't create scene, sceneId is missing.");
+            return null;
+        }
+        return config;
+    }
+
+    @Override
+    public void initialize() {
+        setLocation();
+        logger.debug("Initializing Vera scene handler ...");
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                "Checking configuration and bridge...");
+        mConfig = loadAndCheckConfiguration();
+        if (mConfig != null) {
+            logger.debug("Configuration complete: {}", mConfig);
+            scheduler.schedule(new Initializer(), 2, TimeUnit.SECONDS);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "SceneId required!");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Dispose Vera scene handler ...");
+        if (mConfig.getSceneId() != null) {
+            mConfig.setSceneId(null);
+        }
+        if (pollingJob != null && !pollingJob.isCancelled()) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+        super.dispose();
+    }
+
+    @Override
+    public void handleRemoval() {
+        logger.debug("Handle removal Vera scene ...");
+        scheduler.execute(new Disposer());
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        // Only called if status ONLINE or OFFLINE
+        logger.debug("Vera bridge status changed: {}", bridgeStatusInfo);
+
+        if (bridgeStatusInfo.getStatus().equals(ThingStatus.OFFLINE)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Bridge status is offline.");
+        } else if (bridgeStatusInfo.getStatus().equals(ThingStatus.ONLINE)) {
+            // Initialize thing, if all OK the status of scene thing will be ONLINE
+            scheduler.execute(new Initializer());
+        }
+    }
+
+    private class ScenePolling implements Runnable {
+        @Override
+        public void run() {
+            // logger.debug("Starting polling for scene: {}", getThing().getLabel());
+            for (Channel channel : getThing().getChannels()) {
+                // logger.debug("Checking link state of channel: {}", channel.getLabel());
+                if (isLinked(channel.getUID().getId())) {
+                    // logger.debug("Refresh items that linked with channel: {}", channel.getLabel());
+                    try {
+                        refreshChannel(channel);
+                    } catch (Exception e) {

```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception